src/py/mains/run.py

```diff
@@ -38,7 +38,9 @@ def run_script(job_data, job_type = "SGE" ):
         job_name = job_data["job_name"]
         cwd = job_data["cwd"]
         sge_option = job_data["sge_option"]
-        fc_run_logger.info( "submitting %s for SGE, start job: %s " % (script_fn, job_name) )
+        with open(script_fn, 'r') as original: data = original.read()
+	with open(script_fn, 'w') as modified: modified.write("#!/bin/sh" + "\n" + data)
```

Comment:
I actually tried that at one point with a dummy SH file- fail. Did not like it. Seems like there’s some check in the underlying code looking for that first line and if it’s not there it throws an error, even if you say sbatch bash {script}.

Roberto Lleras
Senior Scientist, Bfx. Field Applications Specialist, North America & Europe
Tel: +1.718.578.8987 | Skype: live:roberto.a.lleras_1 | rlleras@pacb.com<mailto:rlleras@pacb.com>
Pacific Biosciences | 1380 Willow Road, Menlo Park, California 94025 | pacb.com<http://www.pacificbiosciences.com/>

From: Christopher Dunn [mailto:notifications@github.com]
Sent: Monday, August 31, 2015 1:30 PM
To: PacificBiosciences/FALCON
Cc: Roberto Lleras
Subject: Re: [FALCON] Fix support for FALCON 0.3 running on SLURM (#181)


In src/py/mains/run.py<https://github.com/PacificBiosciences/FALCON/pull/181#discussion_r38335805>:

> @@ -38,7 +38,9 @@ def run_script(job_data, job_type = "SGE" ):

>          job_name = job_data["job_name"]

>          cwd = job_data["cwd"]

>          sge_option = job_data["sge_option"]

> -        fc_run_logger.info( "submitting %s for SGE, start job: %s " % (script_fn, job_name) )

> +        with open(script_fn, 'r') as original: data = original.read()

> +     with open(script_fn, 'w') as modified: modified.write("#!/bin/sh" + "\n" + data)

Interesting. I wonder whether we could have used 'bash {script}' on the sbatch line.

Anyway, I'm fine with this -- aside from the tab character.

—
Reply to this email directly or view it on GitHub<https://github.com/PacificBiosciences/FALCON/pull/181/files#r38335805>.